src/mod/irc.mod/msgcmds.c

```diff
@@ -351,7 +351,7 @@ static int msg_who(char *nick, char *host, struct userrec *u, char *par)
   struct chanset_t *chan;
   struct flag_record fr = { FR_GLOBAL | FR_CHAN, 0, 0, 0, 0, 0 };
   memberlist *m;
-  char s[UHOSTLEN], also[512], *info;
+  char s[NICKLEN + UHOSTLEN], also[512], *info;
```

Comment:
NICKLEN + UHOSTLEN + 1 ?

Reply:
In a sane world, yes. But eggdrops usage of UHOSTLEN and friends is borked. This here is an example of something that can be read quite often within eggdrop. Later in the code there is a line like:
`snprintf(s, sizeof s, "%s!%s", m->nick, m->userhost);`
_Note: If egg_snprintf would have been sprintf, the compiler would have catched it. This is what i am trying to to with #1028.
That snprintf cant fit into UHOSTLEN.
There is no better fix currently than oversizing the destination array.
In some distant future i hope to fix the real problem, but that would requite a quite large PR and some tests to make sure its stable. 
OK, in other words, eggdrop needs to define one UHOSTLEN without NICK, and one with NICK. That is currently not the case.

Reply:
To summarize this in one line, m->userhost is defined as UHOSTLEN, even though UHOSTLEN is spaced to allow the nick!user@host format not the user@host format we are expecting. So in theory m->userhost could be longer than expected, even though we only ever write the user@host portion to it